src/api/article/collection/controller.js

```diff
@@ -1,29 +1,21 @@
 import { checkIsFound } from 'utils/validation';
 import { sendJson } from 'utils/api';
 
-import { ArticleCollection, serializeCollection } from './model';
+import { checkPermissions } from 'api/user';
 
-export const getAll = (req, res, next) =>
+import { ArticleCollection, serializeCollection, COLLECTION_POPULATE_OPTIONS } from './model';
+
+export const getAll = ({ user }, res, next) =>
   ArticleCollection.find({ active: true })
-    .populate({
-      path: 'articles',
-      populate: {
-        path: 'locales',
-      },
-    })
+    .populate(COLLECTION_POPULATE_OPTIONS.articles(!checkPermissions(user, ['canManageArticles'])))
```

Comment:
imho we can refactor `checkPermissions` to support both: array & string as second param

Reply:
refactored that, although the only benefit - to be able to avoid `[ ]`